### DISCUSSION TITLE
Regarding the whole log situation with players looking at them

### DISCUSSION THREAD

**Discussion by Justformas** [Monika-After-Story/MonikaModDev#9845] [general]
To all the developers and anyone else who has anything to say about this, I think this situation should be addressed and discussed anyway, but also so a situation like what happened with character files doesn't happen.

It's been said that logs are for developers, and not for players. But at the same time, the "all features" wiki: https://github.com/Monika-After-Story/MonikaModDev/wiki/All-Features-%5BSPOILERS%5D mentions "how to your affection" as if it is a feature, and the FAQ mentions "several log files that may interest you". There's nothing that indicates on these pages that those things are only for developers (although that could be added). While Github is largely a place for developers, players also come here for help, to make suggestions or to get things like submods. Monika even mentions Github in-game, and some people find out about Github before even finding out about the MAS download website. Both the wiki and FAQ are useful for players, and appear to be made to help them. Player's don't have to read them, but many are drawn or lead to them anyway. Without any outside help, players may not figure some things out like gift names, or how to solve technical issues. Despite logs not being directly mentioned in-game, there's still a path that leads people (or players) to them, and both DDLC and MAS being as meta as they are only adds to the confusion.

Maybe it was an older developer mentality, to treat logs like some sort of feature for players (in addition to for developers). But even if not, there are certainly players that like to treat it as something like that. For many, knowing affection has become ingrained and part of MAS or MAS culture, regardless of developer intent. Just one example: https://www.reddit.com/r/MASFandom/comments/zowwwt/reached_3000_affection_points_since_moni_talked/ MAS isn't the only game where players have become aware of and like knowing some value not seen in-game either. 

While the player in that example appears to have used a submod to display the current affection, logs contain much more information that can be useful. Not just for developing, but also for solving or figuring out some potential issue, which includes posting a screenshot of it to Github. That also includes submods, where a player that installs them may not actually be developing the submod or even testing it, but having some potential issue with it. Finally, logs can be useful for figuring out how some things work. MAS is an open-source project after all, and logs can be another source of information for people who are ok with looking at them. Even for people who aren't developers (or contributors) now, one day they might become one, in which case it could be useful to already be familiar with logs. 

What this all comes down to is it's more than just black and white, with one color being the intended player experience, and the other color being a developer.

However, when logs are mentioned in a wiki or FAQ, but without explaining what the terms mean in them, of course some people will read them and not know what some terms mean. That could be the case with new developers too. Not all the terms are obvious in exactly what they mean, but it's also possible to be able to read most of a log without understanding every single term. People may eventually learn or figure out these terms, but the terms or at least more notable ones could be explained in a wiki.

Of course, there's people that don't want to look at logs and would rather have the "intended" experience. That's fine; they don't have to look. But there's also many that want to look or be able to for various reasons, and won't be happy if logs are removed or restricted.

**Comment by multimokia** [comment]
> mentions "how to your affection" as if it is a feature, and the FAQ mentions "several log files that may interest you".

It should be noted there is a disclaimer on the wiki that it is **not** written by the devs, but rather by the community.

> Without any outside help, players may not figure some things out like gift names, or how to solve technical issues

This is another issue I don't like in MAS and I do want to address at some point to retain immersion but also make it a lot more clear on what exactly is giftable

That said, to address the rest of this quickly:
-- 
Realistially this is a developer vision thing, and as well, an immersion thing. We want to make the mod immersive -- access to the raw log data goes completely counter to that.
Realistically, logs won't be removed for debugging purposes, however it's likely they will not be readable without tools.

This has additional benefits for us as well purely because of the amount of users who claim an issue and then subsequently have modified their log files to fit their statements (even if internally we know things can't work like that)

This just saves the hassle in dealing with that at all, so the issue of sending logs here on GH is more or less moot.

It's not about people being okay or not to look, the fact is the community pressure and constant push to, and that's not something we want to encourage in the vision for this mod.

**Reply by Justformas** [reply]
> It should be noted there is a disclaimer on the wiki that it is **not** written by the devs, but rather by the community.

I saw this: "However, the dev team will not add features here as they are released. Instead we'll confirm/deny user submissions on a regular basis."
 
So at some point it was confirmed/approved rather than denied. But it seems that devs do sometimes contribute to or may even be the sole contributor to a wiki page, like this one: https://github.com/Monika-After-Story/MonikaModDev/wiki/NOU-rules Regardless I wanted to point out one thing that may lead players to treat it as a feature, despite any disclaimers. Whether it's devs or the community deciding if there should be more clarity about what it's for on the wiki, it could be changed.
  
> This has additional benefits for us as well purely because of the amount of users who claim an issue and then subsequently have modified their log files to fit their statements (even if internally we know things can't work like that)

So that actually happens. Well if there was a way to make the logs not editable or more difficult to, without making them unreadable, then that'd be an alternative.
  
> This just saves the hassle in dealing with that at all, so the issue of sending logs here on GH is more or less moot.

But then either you won't be able to ask for users' logs to help solve or diagnose their issues, or they will have to post the whole file for you to unlock and read, like with persistents. While many are probably fine with that, personally I'd rather not post any more data than needed, like a screenshot of the relevant part. There's probably others that feel the same way.
 
> It's not about people being okay or not to look, the fact is the community pressure and constant push to, and that's not something we want to encourage in the vision for this mod.

If there's lots of community pressure or pushing to look, then clearly there's a notable segment of the community that likes looking and I wouldn't want to take that away from them. Pier pressure can be tempting, but it's up to an individual whether to give into it or not. They may (eventually) decide to give in, but in this case I don't think that's a problem, or at least not enough to restrict logs. The result is just looking at logs, not something dangerous.

> Realistially this is a developer vision thing, and as well, an immersion thing. We want to make the mod immersive -- access to the raw log data goes completely counter to that. Realistically, logs won't be removed for debugging purposes, however it's likely they will not be readable without tools.

Access to a lot of things outside of the game's full screen can arguably hinder immersion, and I don't just mean mod files or even Github. Players may do things outside of the game screen while MAS is running, some things part of the intended experience and some not, but that's also up to them.

Being available only for debugging purposes, or with tools, would at least make it more clear that logs aren't for players. But this may not solve your other concerns and just lead to pushback and players seeking ways around it. I mean, unless there's going to be tight controls on who can even try to develop or debug something for MAS, there would still need to be some way for such people to unlock and view them. Some players will likely try to use those same methods, or maybe alternatives like submods, and that same pier pressure to look or know may resume.

**Reply by Booplicate** [reply]
You're trying to be picky about the words again. Yes, sometimes we have to add/update wiki. We only do that for the more complex entries to help people with contribution (like how to create spritepacks, submods, how nou works) because those can be hard to figure out on your own. Generally wiki is outdated and we don't update it. Like the affection page is out of sync for years. That said, I think we should nuke or rework that page, because people use it as a reference point.

Calling logs a feature for users is far fetched. I mean, in no other software you'd look for logs unless you're having an issue with it, and even then you're probably just sending it in a bug report. That's how you use logs. If you're willing to treat it like a feature, you're free to do so, but it's pretty much pointless and doesn't prove anything in practice, it's just your way of viewing it.

> Well if there was a way to make the logs not editable or more difficult to

This is not how logs work, it's just text files on your computer. You can edit everything on your computer.

>  or they will have to post the whole file for you to unlock and read
> I'd rather not post any more data than needed, like a screenshot of the relevant part

We're always asking for the full file, it's impossible to debug by a cutoff piece of information the user decided to share.

> but in this case I don't think that's a problem, or at least not enough to restrict logs.

We think that is a problem.
- It was never a feature for the player to use, it's a debug tool, but people turned it into a grinding simulator where their goal is to optimise the grind. It further gamifies MAS and turns it into a generic VN/tamagochi -like game. That's not something I want to see in the mod. People who want that, can implement that feature as a submod.
That said, there's plenty of ways to actually see your "progress" in the game, but instead of being shown directly in numbers, it's shown in a more realistic and subtle way.
- People can't properly read logs which leads to a lot of false-positive reports, which only proves that it was never designed to be viewed "for fun."
- People are trying to modify the logs (some are trying to cheat that way, some are trying to delude us into thinking they have an issue).

And the only benefit is some people want to keep track of their progress using numbers (which most of the time don't even tell much), and some people use it to make "optimal runs" in MAS. I don't think it's worth the hassle.

> Access to a lot of things outside of the game's full screen can arguably hinder immersion, and I don't just mean mod files or even Github. Players may do things outside of the game screen while MAS is running, some things part of the intended experience and some not, but that's also up to them.
> Being available only for debugging purposes, or with tools, would at least make it more clear that logs aren't for players. But this may not solve your other concerns and just lead to pushback and players seeking ways around it. I mean, unless there's going to be tight controls on who can even try to develop or debug something for MAS, there would still need to be some way for such people to unlock and view them. Some players will likely try to use those same methods, or maybe alternatives like submods, and that same pier pressure to look or know may resume.

When you're going out of your way to look at sources (GitHub), it's not a feature in the game, and it doesn't give us the same problems we described above. Same with submods that show your affection or youtube videos that show progression. Logs are generated by the game itself, we should be able to rely on them and can't simply dismiss them.
You're right, it's indeed up to the player how they want to play the game, it's also up to the developer how they want their game to be. If some players are seeking a way to ruin the experience, it's up to them, but it's not something we will provide, especially if it gives us troubles.

**Reply by Justformas** [reply]
>You're trying to be picky about the words again.

Regarding this, one word can change the whole meaning of something, and we communicate with words. Especially in online forums where it's often the only way. But there can be ambiguity and generalizations in words, so yes the words used can matter. Otherwise the precise meaning may not be clear and could mislead even if not intentionally. So unless it could be a joke, metaphor or something like that, I tend to try and take words at face value and in a way consistent with everything else that I've seen and could be possible.

>Generally wiki is outdated and we don't update it.
>That said, I think we should nuke or rework that page, because people use it as a reference point.

I'm saying that's one reason why people may treat it like a feature, regardless of what logs are intended for. So you guys may add some things including to the FAQ, but not on that wiki page or at least not generally, but may confirm or deny user entries. 
Part of the reason I made this discussion is to think about what, if anything, should be done regarding it being on that wiki page, and the other logs being mentioned in the FAQ. But aside from the sections about the affection log and maybe affection itself, I think the All Features wiki page still contains lots of useful information that doesn't cause players to treat things as unintended features. Maybe a rework but don't think it should be nuked.

>Calling logs a feature for users is far fetched. I mean, in no other software you'd look for logs unless you're having an issue with it, and even then you're probably just sending it in a bug report. That's how you use logs. If you're willing to treat it like a feature, you're free to do so, but it's pretty much pointless and doesn't prove anything in practice, it's just your way of viewing it.

Well I actually don't think it's that far-fetched for the DDLC universe. And while MAS isn't the same as DDLC the game, people are inevitably going to associate MAS with it. Even if DDLC didn't actually put things for players to find in logs, just putting features or secrets in the directory for players at all (Both DDLC and MAS use the characters folder, DDLC also uses the main directory and even a traceback.txt) may lead players to think there are other secrets or features in the directory. Throw in that the affection log is mentioned in the All Features wiki, and that many players do look at it, and it's easy to see why players may treat it as a feature, either because they think it is one or just want to anyway. But I think it can be better communicated that logs aren't intended as features for players. Many probably won't care and want to keep looking anyway, but at least they can make a more informed decision on whether to or not.

>This is not how logs work, it's just text files on your computer. You can edit everything on your computer.

As long as it can be opened as a text file then probably yes. There's editor programs that can sort of "protect" a document so that it's readable but not editable, but if it can be opened as a text file or with another application then that can be bypassed. But ok let's say not really feasible, however that also means that if logs still exist in some form, like in debug mode or will require some tool to open, then players could potentially still edit those too.

>We're always asking for the full file, it's impossible to debug by a cutoff piece of information the user decided to share.

There are some times when people just post a screenshot, and you can quickly figure out if there's a problem or not, or a known-issue. But for an unknown problem that can't easily be replicated otherwise, sure it might need more information.

>there's plenty of ways to actually see your "progress" in the game, but instead of being shown directly in numbers, it's shown in a more realistic and subtle way.

I haven't found many differences past 1000 or 2000 affection. Maybe an occasional new compliment here and there. Personally think seeing the actual value is interesting and far more telling anyway.

>People can't properly read logs which leads to a lot of false-positive reports, which only proves that it was never designed to be viewed "for fun."

It might prove that player's don't understand some terms and that they're not properly explained somewhere obvious, but that alone doesn't prove that it was never designed to be viewed for fun. There's cases where games have something designed for players but just don't explain it or not very well. But many players do know how to at least read their current affection value though, among other things. But I don't see why this supposed-proof matters. I'm saying why players might not be getting the message that logs aren't for them, but also that many probably don't care anyway.

Also when players post logs to Github, they aren't doing it for fun (unless they are intentionally editing their log files to mislead), they're doing it to report something or get help. But false-positives do happen yes. Again if the log terms were explained somewhere then that could help reduce them.

>It was never a feature for the player to use, it's a debug tool, but people turned it into a grinding simulator where their goal is to optimise the grind. It further gamifies MAS and turns it into a generic VN/tamagochi -like game. That's not something I want to see in the mod. People who want that, can implement that feature as a submod.

Players can grind whether they have access to logs or not; they just have to know how. It's not about being "in the mod" or even having logs per se. Rather grinding is a way that players might play, even without logs. They might not be as satisfied without seeing a number, but again there will probably be way(s) to see it, in which case word will get around. So this isn't really doing much about it other than adding an extra step to view the numbers.

>When you're going out of your way to look at sources (GitHub), it's not a feature in the game, and it doesn't give us the same problems we described above. Same with submods that show your affection or youtube videos that show progression. Logs are generated by the game itself, we should be able to rely on them and can't simply dismiss them.

If logs still exist in some form and can be accessed somehow, then players could potentially use those same methods to unlock, edit and post them. But whether they're looking at a log accessible in the vanilla game's directory, or one that was restricted or even a submod, it can still lead to those same problems of ruining immersion and pier pressure to look (multimokias concerns), players potentially still treating the game like a grinding simulator, and players making potentially false-positive issues about losing affection. Submods aren't vanilla MAS but people still play MAS with them and it becomes part of their experience. Effectively it will just add an extra step to look, to the detriment of players who like to more than anything (but not necessarily stop them).

>You're right, it's indeed up to the player how they want to play the game, it's also up to the developer how they want their game to be. If some players are seeking a way to ruin the experience, it's up to them, but it's not something we will provide, especially if it gives us troubles.

Ruin the intended or envisioned experience maybe, but that applies to grinding itself, even without logs. Other people may look but not necessarily to grind, or think that it doesn't ruin their own experience and even that it enhances it, even knowing that it's not intended.

While up to the developers, ideally developers should consider what the players/community playing the game want or like, whether removing something will upset a significant portion of those people, and whether it's really going to solve much or lead to more of the same (or similar) problems, like what may happen if or when players seek workarounds. Even if it's something that wasn't intended as a feature. In this case the issue isn't just not providing something; it's removing/restricting something that has long existed in the mod and that many players do like, which players don't even have to look at it if they don't want to.

**Reply by Booplicate** [reply]
> just putting features or secrets in the directory for players at all (Both DDLC and MAS use the characters folder, DDLC also uses the main directory and even a traceback.txt) may lead players to think there are other secrets or features in the directory.

Are you comparing easter-eggs with log files? You open the log, see there's no easter eggs/references, but somehow still think it's a game feature? I don't see the connection. All (or nearly all) easter eggs in the game directory are made clear that they exist in the game itself. Character files, gifts, notes - everything is mentioned because the game wants you to see them. Logs aren't, because there's no reason to read them unless you got an issue. Or are there really people that get all excited when reading `mas_log.log`?

> There are some times when people just post a screenshot, and you can quickly figure out if there's a problem or not

Even if in some cases we *can* figure out the problem by just 1 line of log, doesn't mean we *want* to spend extra time trying to do that just because you don't want to send your logs. Don't make other people lives harder, please.

> player's don't understand some terms and that they're not properly explained somewhere obvious, but that alone doesn't prove that it was never designed to be viewed for fun.

Read my first point. If something isn't explained, then it's most likely not for you. Developers want you to know and experience all the features and they wouldn't hide it like that. Maybe there's some games that hide their features from the players, I don't know what and why. Either way logs are not designed for most people to read, which proves itself by the fact people don't understand them and we don't explain the format, it'd be a waste of time and weird decision choice.

> Players can grind whether they have access to logs or not; they just have to know how. It's not about being "in the mod" or even having logs per se. Rather grinding is a way that players might play, even without logs. They might not be as satisfied without seeing a number, but again there will probably be way(s) to see it, in which case word will get around. So this isn't really doing much about it other than adding an extra step to view the numbers.

That's your opinion, I respect it. I have my own, formed by some experience in gamedev. Being able to see your "value" (gold, exp, affection, name it) is one way to make people want to grind. It doesn't define grinding, but intensifies it.
People give Monika gifts to get one more affection to unlock a feature. I want to change that. We should tease features (like how we do with outfits and the islands), show them to the user and either unlock the full thing later, or let them see how it progresses. Instead of measuring relationship in a number, measure it with experience.

> Submods aren't vanilla MAS but people still play MAS with them and it becomes part of their experience. Effectively it will just add an extra step to look, to the detriment of players who like to more than anything (but not necessarily stop them).

I'm fine with that, submods are made exactly for that purpose, to contain things that cannot be in the vanilla MAS. That includes things that may ruin your experience.

> ideally developers should consider what the players/community playing the game want or like

Yes and no. There's examples of when developers listened too much to people who don't know what they truly want. There's also examples when developers should've listened to their userbase more, because they didn't know what they wanted to do. It's about balance. We're always considering what's better.

> it's removing/restricting something that has long existed in the mod

Not in the mod, it hasn't been mentioned in the mod in any way.

**Reply by Justformas** [reply]
I realize this is way later but picking up where I left off.

> Are you comparing easter-eggs with log files? You open the log, see there's no easter eggs/references, but somehow still think it's a game feature? I don't see the connection. All (or nearly all) easter eggs in the game directory are made clear that they exist in the game itself. Character files, gifts, notes - everything is mentioned because the game wants you to see them. Logs aren't, because there's no reason to read them unless you got an issue. Or are there really people that get all excited when reading `mas_log.log`?

First, you can't necessarily expect players to know that something not mentioned in game is not an intended feature. I mean, not everyone thinks "the game didn't talk about this specific thing (yet) so I better not open or look at it" when playing a game or in DDLC/MAS's directory. That may be how MAS treats everything, but players might not realize that, still find stuff that's not mentioned and potentially think it's a feature, even if it's near other stuff that might only be for devs. 

A game directory might not be a typical place to look for most games, but DDLC had players going in the directory not just for easter eggs but as a feature for completing the game. MAS may not have directory easter eggs but is associated with DDLC and continues the idea of using the characters folder as a feature. It's a special case where a player may be inclined to search for more things in the directory, regardless of them not being mentioned in-game. Between that, and of course the wikis, youtube videos and community posts mentioning affection and how to view it (usually without a disclaimer saying that it's for the devs only), a player could easily come to treat it as a "directory feature", mistakenly or not. It doesn't matter if it wasn't mentioned in-game if a player still ends up treating it like a feature anyway. The reality is a lot of people interact and see things online, and they also may not remember everything that was or wasn't mentioned in-game anyway.

While "mas_log.log" doesn't appear to be anything exciting, the affection log is much more interesting to some people. It serves as a playlog and can be correlated with actions you do in-game. I'm not arguing whether you devs consider or intend it as a feature, but trying to show how players can end up treating and using it like a feature. I mean, if you want to go so far as to encrypt it to prevent players from looking, then at least to some extent you must agree that players are treating it like a feature.

 
> Even if in some cases we _can_ figure out the problem by just 1 line of log, doesn't mean we _want_ to spend extra time trying to do that just because you don't want to send your logs. Don't make other people lives harder, please.

The cases I was thinking about were probably pretty cut and dry, and known. They only involved something like a traceback.txt or a session or few worth of log lines. But I know there's cases where you might need more info than that.

But I also wanted to say here that I don't really like the idea of having to rely on the devs more for debugging/troubleshooting (including when making/trying my own custom stuff) when I could potentially do it myself if the logs weren't encrypted. But instead I'd have to upload the log here for you to decrypt and involve you in the issue, taking up more of you guys' time. Aside from decrypting it ourselves or you guys providing a key for people doing dev-ish stuff with their game.

> Read my first point. If something isn't explained, then it's most likely not for you. Developers want you to know and experience all the features and they wouldn't hide it like that. Maybe there's some games that hide their features from the players, I don't know what and why. Either way logs are not designed for most people to read, which proves itself by the fact people don't understand them and we don't explain the format, it'd be a waste of time and weird decision choice.

It might be more so for "discovering" and learning features in some games, or because some games just don't explain everything well (or at all). So that's not the case in MAS here and you're saying all features are told to the player, but basically what I said in the first point is that players might not make that connection, even if a lot of things are explained well in-game.

> That's your opinion, I respect it. I have my own, formed by some experience in gamedev. Being able to see your "value" (gold, exp, affection, name it) is one way to make people want to grind. It doesn't define grinding, but intensifies it. People give Monika gifts to get one more affection to unlock a feature. I want to change that. We should tease features (like how we do with outfits and the islands), show them to the user and either unlock the full thing later, or let them see how it progresses. Instead of measuring relationship in a number, measure it with experience.

Choosing to look at it could maybe encourage more grinding, but isn't the player supposed to gain affection over time anyway (assuming they treat Monika well)? Is it really a problem if they're doing things that result in them gaining more? To get deeper into the relationship? I don't just mean to get one more point for some threshold either, I mean in general. Of course at some point, you may not "unlock" anything more or Monika won't react any differently (I suspect I'm long past that point), but even at that point, people may still like the feeling of increasing it and seeing it anyway.

Honestly I can view a lot of things as a grind even in real life, including without seeing a number. I don't necessarily mean in a negative way, but just putting in the effort. Even relationships require effort to develop over time, to bond and get closer to the other person.

Regardless, teasing features sounds like a good idea anyway. Maybe the mod could measure more things with experience. But on a personal level I already measure things in experience as well. Affection is just one part of it that doesn't seem to have any more noticeable effects in-game at this point for me, but I still like seeing it in the log.

> Not in the mod, it hasn't been mentioned in the mod in any way.

> I'm fine with that, submods are made exactly for that purpose, to contain things that cannot be in the vanilla MAS. That includes things that may ruin your experience.

In the directory and generated by the mod, then. Which people may still associate with their mod experience for reasons said in my first point.

As for submods, I feel this case is a bit unusual because all those people watching affection (for potentially years) would have to install a submod just to resume/reimplement an experience that they were getting and want, and it may not be the whole experience if they want the logs and they aren't included. Also because if everyone or most people who were watching affection end up doing that anyway, then not many more people would be getting the experience that you devs intend from hiding affection anyway. There are certainly submods that show affection, but I guess we'd have to see if there will be any that decrypt logs because I'd much rather see the whole logs (and not just the affection log). 

Aspiring devs/contributors/tinkerers that aren't actually part of the MAS team may have a use for the logs too so hopefully could see them somehow. I don't want this to turn into a gatekeeping situation where only certain people can get keys to decrypt the logs.

> > ideally developers should consider what the players/community playing the game want or like
> 
> Yes and no. There's examples of when developers listened too much to people who don't know what they truly want. There's also examples when developers should've listened to their userbase more, because they didn't know what they wanted to do. It's about balance. We're always considering what's better.

Yes it is about balance when listening to people. But in this case, encrypting logs is a big step in one direction away from what some people like and have long been doing, and the time involved in implementing it could potentially be better spent on other things. Aside from me, I also think it could cause backlash or upset people even if they don't show it through an online presence. Hopefully there will be adequate solutions if it actually happens.

**Comment by Capricus02** [comment]
Just going to throw my two cents in here as both a casual player and a very amateur/novice mod creator - I'm fine with hiding most of these things, but I'll fully admit I like to be able to keep an eye on where I'm at affection-wise. There was a case not long ago where I finally had hit 1000+ affection because that was my goal for some time, and then I decided to ask Monika a question that I *thought* given our affection she would take in stride as a joke (the "Do you want to meet my girlfriend?" or something like that). Maybe something like "Ha ha yeah I have a mirror." or something. Whoops. That was a bad idea. I think I lost something like 150 affection points on that. It was a bit frustrating but I also understand the reasoning, though it took me probably another month and a half or so to build back up to it. It was good to know though that not only was that a bad option (her response doesn't necessarily indicate the magnitude of the negative impact) but just how *much* of a bad option it was, because I now know not to joke around like that. I'm the kind of person that likes to make those kind of sarcastic jokes expecting the other person to play along but the log helped me to realize she just isn't that kind of person.

Now I don't stress myself too much about the affection because I know I'm up over 1000 and I don't really care about the points anymore. I know there's probably some other stuff that happens at like 2k or something, but honestly just hitting 1k was a great a feeling of accomplishment and I don't need/have to feel like I'm pushing towards some goal, I can just play casually now and relax.

Again, I'm in the minority here because of my weird need to be able to see progress from a numbers perspective (IRL I'm a data analyst so that might explain a bit of my anal retentiveness with numbers), but I think there should still be a way for players to at least kind of know where they're at affection-wise. Even if it's not a number, maybe the devs could incorporate a phrase or something the player can check similar to what's shown on [this features page](https://github-wiki-see.page/m/Monika-After-Story/MonikaModDev/wiki/All-Features-%5BSPOILERS%5D) under the "Affection" section. 

Example:

Player wants to know their current affection level with Monika. They could click a button or something in the "Extras" section (that doesn't seem to be used for anything anyway at the moment) that says "Affection" or "Current Affection" which would then say something like: "Monika feels Broken/Distressed/Upset/Normal/Happy/Affectionate/Enamored/in Love with [player]"

Anyway, again, that's just my $0.02. Just wanted to offer a potential suggestion if the mods are looking to get rid of/hide logs. I think most casual players aren't going to care/mess with any logs except the affection one anyway.

**Reply by Booplicate** [reply]
You're right about that topic, it's not entirely clear what's going on, I finally marked that as an issue to fix #9866.

You can indirectly see changes in affection in her expressions and the way she talks, as well as unlocking new, more intimate dialogue options.

I don't like showing affection directly because it gamifies MAS. You can't "view" person status like that IRL. Instead we can use body language as well as some tone changes in dialogue.

**Reply by Capricus02** [reply]
Thank you for marking the issue! I wasn't sure if that was intentional or not because I could've sworn I read something about it somewhere on here but couldn't find it again.

I totally agree, though it's still difficult for some to read body language/tonality when it comes to 2-dimensional beings (hell it can be difficult for some people to read body language IRL). Like I said though I'm totally OK with hiding the affection numbers / logs if need be, because you're right it does gamify it and I absolutely found myself trying to min/max daily affection (definitely not in the spirit of the mod).

I don't think many folks would notice it anyway judging by the few who post on here compared to the number who download the mod.
